packages/suite/src/storage/migrations/index.ts

```diff
@@ -781,4 +783,96 @@ export const migrate: OnUpgradeFunc<SuiteDBSchema> = async (
         // @ts-expect-error
         backendSettings.delete('trop');
     }
+
+    if (oldVersion < 37) {
+        const accountsStore = transaction.objectStore('accounts');
+        await accountsStore
+            .openCursor()
+            .then(async function updateMetadataKeys(cursor): Promise<void> {
+                if (!cursor) {
+                    return;
+                }
+                const account = cursor.value;
+
+                // @ts-expect-error
+                if (!account.metadata?.fileName) {
+                    return; // is this ok? return here?
```

Comment:
I believe this `return` will stop the recursion immediately and thus leave the rest of accounts unmigrated. I would negate the condition and move the `account.metadata` transformation and `cursor.update` inside the block.